README.md

```diff
@@ -1,2 +1,146 @@
-# marmopy-sdk
-Marmo wallet Python SDK
+# Marmopy-sdk
+
+# Description
+Marmopy-sdk helps python developers to consume Marmo Relayer.
+
+# Simple Summary
+Allowing users to sign messages to show intent of execution, but allowing a third party relayer to execute them is an emerging pattern being used in many projects. 
+This pattern simplifies the integration with any Ethereum based platform. 
+Marmo relayer: (https://github.com/ripio/marmo-relayer). 
+
+# Abstract
+### User pain points:
+> - Users don't want to think about ether
+> - Users want to be able to pay for transactions using whatever they have 
+> - Users don’t want to download apps/extensions (at least on the desktop) to connect to their apps
+
+# Ecosystem Graph
+![](./images/01.png)
+
+###### General WIKI ecosystem.
+- Work in progress.
+###### API layer
+- Marmo relayer doc: (https://github.com/ripio/marmo-relayer/blob/master/README.md)
+###### CORE layer
+- Marmo contracts doc: (https://github.com/ripio/marmo-contracts/blob/master/README.md)
+
+# Features
+- Complete implementation of Intent functionality for Marmo relay.
+- Ethereum wallet support.
+- Comprehensive integration tests demonstrating a number of the above scenarios.
+
+##### Runtime dependencies:
+
+- web3py (https://pypi.org/project/web3/).
+
+# Getting started
+
+##### Prerequisites
+* Python3
+
+Clone the repository and then run:
+
+pip install .
+
+
+### Intent Flowchart
+![](./images/02.png)
+
+###### Dependencies
+> - T0  -> -
+> - T1  -> - 
+> - T2  -> T1
+> - T3  -> T1
+> - T4  -> T1
+> - T5  -> T1, T4
+> - T6  -> T2
+> - T7  -> T2
+> - T8  -> T2
+> - T9  -> T3
+> - T10 -> T4
+> - T11 -> T5
+> - T12 -> T6
+> - T13 -> T7
+> - T14 -> T8
+> - T15 -> T9
+> - T16 -> T10
+> - T17 -> T10
+> - T18 -> T11
+> - T19 -> T13
+> - T20 -> T14
+> - T21 -> T15
+> - T22 -> T18
+> - T23 -> T19
+> - T24 -> T21
+> - T25 -> T22
+> - T26 -> T12, T23, T20, T15
+> - T27 -> T24
+> - T28 -> T25
+> - T29 -> T27
+
+
+### Build a intent
+```python
+from marmopy import Intent,Credentials,IntentAction
+
+class Contract:
+    def __init__(self,contractAddress):
+        self.contractAddress = contractAddress
+        
+    @IntentAction
+    def transfer(self, to:'address', value:'uint256') -> 'bool': pass
+    
+    
+tokenContractAddress = "0x2f45b6fb2f28a73f110400386da31044b2e953d4" #RCN TOKEN
+to = "0x7F5EB5bB5cF88cfcEe9613368636f458800e62CB"
+
+erc20 = Contract(tokenContractAddress)  #example of ERC20 contract with IntentActions injections. 
+
+value = 1
+
+intentAction = erc20.transfer(to,value)
+
+credentials = Credentials("Your private key here");
+
+contractAddress = "0x7c0e959d9b7c82d47c5013832f89282b8b773393"
```

Comment:
This is the address of the Marmo wallet, it could be generated directly inside the SDK, something like `marmoOf(Credentials("Your private key here")) = 0x7c0e959d9b7c82d47c5013832f89282b8b773393`

Reply:
Exactly, the user would need to provide the address of the Marmo creator.

Reply:
Sounds good, do you have the address of the Marmo creator of this wallet: "0x7c0e959d9b7c82d47c5013832f89282b8b773393"  so I can test my logic?

Reply:
This is the creator https://ropsten.etherscan.io/address/0x2e8605b785ef1570e853070636f0582a8900975c#code